hazelcast/src/main/java/com/hazelcast/query/impl/InternalIndex.java

```diff
@@ -118,10 +121,19 @@ public interface InternalIndex extends Index {
 
     /**
      * @return Step-aware storage that backs the Index.
-     * By default returns {@code null} that indicates there is no
-     * Step-aware backed storage.
+     * By default, returns {@code null} that indicates
+     * there is no step-aware backed storage.
      */
     default StepAwareStorage getStepAwareStorage() {
         return null;
     }
+
+    /**
+     * @param isDescending whether the index is used in descending order.
+     * @return corresponding comparator for this index.
+     */
+    @Nullable
```

Comment:
why is it nullable?

Reply:
the name and documentation is misleading. It it not the comparator you would expect on index (comparing values in the index), but comparator ordering IMap keys stored for given index key

Reply:
> why is it nullable?

Well, it was a hard choice between returning null and throwing UnsupportedException in non-ordered index stores.

Reply:
https://github.com/hazelcast/hazelcast-mono/pull/52/commits/b3cf0a3cda4f822488c1b530485969b050741606
https://github.com/hazelcast/hazelcast-mono/pull/52/commits/c2b71cd2bc5489bd928ba6224fd896132797f66b

Reply:
✔️